src/ui/components/button/BaseButton.tsx

```diff
@@ -0,0 +1,39 @@
+import { PropsWithChildren } from 'react';
+import { baseButtonStyle } from '@ui/components/button/button.css';
+import styleToken from '@ui/core/styleToken.css';
+
+type Props = {
+  className?: string;
+  color?: keyof typeof backgroundColorStyle;
+  onClick: () => void;
+  [key: string]: any;
```

Comment:
```suggestion
type ButtonColor = 'primary' | 'secondary' | 'default';

interface ButtonProps {
  className?: string;
  color?: ButtonColor;
  onClick: () => void;
}

type Props = ButtonProps & React.ButtonHTMLAttributes<HTMLButtonElement>;
```

P2: 조금더 확장 가능한 형태로 바꿔보았어요. 이 컴포넌트의 본질은 button이기에 button에서 발생할 수 있는 props를 병합하면 규격화된 형태에서 타입을 확장할 수 있을 것 같아요!

Reply:
> P2: 조금더 확장 가능한 형태로 바꿔보았어요. 이 컴포넌트의 본질은 button이기에 button에서 발생할 수 있는 props를 병합하면 규격화된 형태에서 타입을 확장할 수 있을 것 같아요!

React.ButtonHTMLAttributes<HTMLButtonElement> 같은 속성이 있는지 모르고 있었는데, 너무 도움 되었습니다. 👍

그리고 위에 리뷰주신 ButtonColor 같은 경우에 궁금한 부분이 있는데요.

```ts
const backgroundColorStyle = {
  primary: styleToken.color.primary,
  secondary: styleToken.color.secondary,
  default: styleToken.color.gray['300'],
};

type ButtonColor = keyof typeof backgroundColorStyle;
type ButtonColor = 'primary' | 'secondary' | 'default';
```

가이드 주신 키 값을 직접 명시적으로 적어주는것이 더 올바른 방법일지 궁금합니다.
명시적으로 적어두면 어떤 타입을 사용하는지 바로 확인이 가능해서 가독성이 좋을것 같은데,

`keyof typeof` 를 사용하면 backgroundColorStyle 에 값을 추가하더라도 자동으로 타입이 반영되어서 어떤 부분이 더 괜찮은지 항상 고민되긴 했습니다!